### DISCUSSION TITLE
Changes to Pre-Request Variable Handling in v1.28.0: From Type Inference to Strings

### DISCUSSION THREAD

**Discussion by helloanoop** [usebruno/bruno#3072] [Announcements]
In earlier versions of Bruno, we supported type inference on variables, allowing developers to use JavaScript template literals inside request variables. These were evaluated during runtime, providing flexibility in handling different data types.

With the release of v1.28.0, we’ve shifted to treating all Pre-Request Vars as strings. This change aligns Pre-Request Vars with how environment variables, runtime variables, and folder variables are handled, ensuring a more consistent and predictable behavior across the client.

For documentation around Variables - please visit https://docs.usebruno.com/get-started/variables/overview

**Comment by martinsefcik** [comment]
If I understood it correctly using JavaScript template literals are not supported in 1.28.0+ for request variables at all. Then shouldn't the following hint be also removed from UI ? Otherwise it is confusing.
![image](https://github.com/user-attachments/assets/2f3db79a-f656-4415-99bd-7a865c4d3db0)

Also wouldn't it be better to mention/highlight such backward incompatible changes in the release notes ?
In recent time when I'm updating Bruno I have to often spent hours with investigating why my collection stop working and I have to dig in Bruno source code and identify what was changed and why something what was working correctly in old version is not working in new version.

**Reply by Its-treason** [reply]
Yeah, I'm also unsure, why it got removed. I understand that the String to Number casting was unexpected. But the string literals were a pretty good feature to create values on the fly.

**Reply by helloanoop** [reply]
Hey @martinsefcik 

> Then shouldn't the following hint be also removed from UI ? Otherwise it is confusing.

This has been removed (I believe `v1.32.0` onwards)

> Also wouldn't it be better to mention/highlight such backward incompatible changes in the release notes

This has been highlighted in the downloads page as well as in the [changelog](https://www.usebruno.com/changelog),  page
<img width="747" alt="image" src="https://github.com/user-attachments/assets/9e3f7674-88e4-463d-9d59-40c32f348607">

Unfortunately, This was missed in the release notes. I have added it [now](https://github.com/usebruno/bruno/releases/tag/v1.28.0)

> In recent time when I'm updating Bruno I have to often spent hours with investigating why my collection stop working

Sorry for the trouble there @martinsefcik

**Comment by TysonK5** [comment]
@helloanoop I love your product but I do have questions about this change. 

To preface, Im not a dev but I do work on a software team and Im learning.

I have been learning JS by writing it in the scripting for postman and now bruno but with the recent changes I can no longer simply pass an envVar in the request of my graphQL or rest requests....  

first case I was using momentJS to set dates dynamically based when requests to mimic a date dropdown in the app such as 
- last 30 days
- last month
- last year
- this year

I was able to pass the values into the req from my prescript using moment  js in the past but now they are passing the varible value as a string? {{last30days}} 

how can I fix this?

**Comment by haodeon** [comment]
Can't we have both?

Enable request vars to have type inference on a per variable basis through a checkbox or something?

It was a useful feature and a differentiator to Postman.
